internal/ingress/controller/config/config.go

```diff
@@ -715,7 +715,7 @@ func NewDefault() Configuration {
 		UseGeoIP:                         true,
 		UseGeoIP2:                        false,
 		WorkerProcesses:                  strconv.Itoa(runtime.NumCPU()),
-		WorkerShutdownTimeout:            "10s",
+		WorkerShutdownTimeout:            "300s",
```

Comment:
This is too long for a default value

Reply:
> This is too long for a default value. I think we should leave its default to whatever Nginx sets it to.

I don't think that's the "right" value. The default value for `terminationGracePeriodSeconds` is 30 seconds. If we use NGINX default value of `WorkerShutdownTimeout=10s` is too low?

Just to be clear, `terminationGracePeriodSeconds` is the upper limit, after that time, the pod is destroyed. Using `WorkerShutdownTimeout=300s` doesn't means we will spend five minutes. After the termination signal, NGINX will stop accepting new connections. For instance, if it takes 45s to finish the pending connections, the pod will be terminated just after that, not 300s.

Makes sense?

Reply:
I don't have an issue with bumping `terminationGracePeriodSeconds`. But setting `WorkerShutdownTimeout` to `300s` means in case of a sticky client if there are reloads Nginx workers will pile up and potentially cause OOMKill. Are you bumping that value in response to an issue? I noticed that by default Nginx does not even set that, meaning no timeout I think. Can we at least set that to `240s` to give Nginx a minute to shut down its workers in its own way instead of having kubelet to kill the whole pod.